saleor/dashboard/order/views.py

```diff
@@ -614,14 +629,18 @@ def fulfill_order_lines(request, order_pk):
                 'Fulfilled %(quantity_fulfilled)d items',
                 'quantity_fulfilled') % {
                     'quantity_fulfilled': quantity_fulfilled}
-            order.history.create(content=msg, user=request.user)
+            order.events.create(
+                parameters={'quantity': quantity_fulfilled},
+                user=request.user,
+                type=OrderEvents.FULFILLMENT_FULFILLED_ITEMS.value)
             if form.cleaned_data.get('send_mail'):
                 send_fulfillment_confirmation.delay(order.pk, fulfillment.pk)
-                send_mail_msg = pgettext_lazy(
-                    'Dashboard message related to an order',
-                    'Shipping confirmation email was sent to user'
-                    '(%(email)s)') % {'email': order.get_user_current_email()}
-                order.history.create(content=send_mail_msg, user=request.user)
+                order.events.create(
```

Comment:
What do you think about writing the event in the celery task? When it is done? As the mail can fail to be sent and raise an exception for example.

Reply:
Initially, it will be possible to resend the email  directly from the event, in case of error event won't be added at all.
I think the solution might be adding a status to the event (like STARTED/ERROR/FINISHED) but for now I think it's out of scope